src/coreComponents/finiteElement/elementFormulations/Qk_Hexahedron_Lagrange_GaussLobatto.hpp

```diff
@@ -377,6 +403,39 @@ class Qk_Hexahedron_Lagrange_GaussLobatto final : public FiniteElementBase
                         real64 const (&X)[numNodes][3],
                         FUNC && func );
 
+  /**
+   * @brief computes the non-zero contributions of the d.o.f. indexd by q to the
+   *   stiffness matrix R for the elastic case, i.e., the superposition matrix of first derivatives
+   *   of the shape functions.
+   * @param q The quadrature point index
+   * @param X Array containing the coordinates of the support points.
+   * @param func Callback function accepting three parameters: i, j and R_ij
+   */
+  template< typename FUNC >
+  GEOSX_HOST_DEVICE
+  static void
+  computeFirstOrderStiffnessTerm( int q,
+                                  real64 const (&X)[numNodes][3],
+                                  FUNC && func );
+
+
+  /**
+   * @brief evaluate the gradient of a basis function at a given point
+   * @param q The quadrature point index
+   * @param X Array containing the coordinates of the support points.
+   * @param coords the coordinate of the point where the gradient will be applied
+   * @param func Callback function accepting parameters: the gradient vector and the Jacobian
+   */
+  template< typename FUNC >
+  GEOSX_HOST_DEVICE
+  static void
+  evaluateGradient( int q,
+                    real64 const (&X)[numNodes][3],
+                    real64 const (&coords)[3],
+                    FUNC && func );
```

Comment:
Is `evaluateGradient` used / going to be used anywhere? I did not find it in the solver code (not asking you to remove it, just trying to understand)

Reply:
Rha I forgot that we change the way to compute the sources and so we do not use this function anymore. Thanks I'll remove it.